### PR TITLE
NONJAVACLI-3460: update dependencies

### PR DESCRIPTION
**Pull request by janjwerner-confluent**
Continuation of https://github.com/confluentinc/librdkafka/pull/4690

### CODE REVIEW

File: mklove/modules/configure.libcurl [hunk 1]
```diff
@@ -45,8 +45,8 @@ void foo (void) {
 function install_source {
     local name=$1
     local destdir=$2
-    local ver=7.86.0
-    local checksum="3dfdd39ba95e18847965cd3051ea6d22586609d9011d91df7bc5521288987a82"
+    local ver=8.7.1
+    local checksum="f91249c87f68ea00cf27c44fdfa5a78423e41e71b7d408e5901a9896d905c495"
```
Comment (emasab) [inline mklove/modules/configure.libcurl:49]:
```suggestion
    local ver=8.8.0
    local checksum="77c0e1cd35ab5b45b659645a93b46d660224d0024f1185e8a95cdb27ae3d787d"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,11 @@ librdkafka v2.4.0 is a feature release:
  * Fix for an undesired partition migration with stale leader epoch (#4680).
  * Fix hang in cooperative consumer mode if an assignment is processed
    while closing the consumer (#4528).
+ * Upgrade OpenSSL to v3.0.13 (while building from source) with various security fixes,
+   check the [release notes](https://www.openssl.org/news/cl30.txt)
+   (@janjwerner-confluent, #4690).
+ * Upgrade zstd to v1.5.6, zlib to v1.3.1, and curl to v8.7.1 (@janjwerner-confluent, #4690).
```
Comment (milindl) [inline CHANGELOG.md:28]:
```suggestion
 * Upgrade zstd to v1.5.6, zlib to v1.3.1, and curl to v8.8.0 (@janjwerner-confluent, #4690).
```

**Review by emasab — APPROVED** [review]
LGTM! Thanks

**Comment by glansbur** [comment]
Thank you @emasab

**Pull request merged**